scripts/iso-manager.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+BOOTSTRAP_CONFIG="/etc/metalk8s/bootstrap.yaml"
+DEBUG="info"
+LOGFILE="/tmp/iso.log"
+DRY_RUN="False"
+
+# Helpers
+_usage() {
+echo "iso-manager.sh [options]"
+echo "Options:"
+echo "-p/--product:     Path to product folder or ISO"
+echo "-d/--dry-run:     Run actions in dry run mode"
+echo "-l/--log-file:    path to log file"
+}
+
+# Exit with error code and message
+die() {
+    echo "$*" 1>&2 ;
+    exit 1;
+}
+
+# Run commands and log their output
+_log() {
+    if [ -n "$LOGFILE" ]; then
+        "$@" >> "$LOGFILE" 2>&1
+    else
+        "$@" >&2
+    fi
+}
+
+# Run commands and display/write the output to stdout/logfile
+_info() {
+    if [ "$DEBUG" == "info" ]; then
+        if [ -n "$LOGFILE" ]; then
+            "$@" >> "$LOGFILE" 2>&1
+        else
+            "$@" >&2
+        fi
+    else
+        "$@" >&/dev/null
+    fi
+}
+
+# helper function to set the current saltenv
+_set_env() {
+    if [ -z "$SALTENV" ]; then
+        SALTENV="metalk8s-$(salt-call --out txt slsutil.renderer \
+            string="{{ pillar.metalk8s.nodes[grains.id].version }}" \
+            | cut -c 8-)"
+    fi
+}
+
+# helper function to check for element in array
+containsElement () {
+  local e match="$1"
+  shift
+  for e in "$@"; do
+      [[ "$e" == "$match" ]] && return 0;
+  done
+  return 1
+}
+
+_add_products() {
+    # Use salt file.serialize merge require having full list
+    products=($(salt-call pillar.get metalk8s:products \
+        --out txt | cut -d' ' -f2- | tr -d '[],'))
```

Comment:
Having an output example in comment would help to understand the "parsing" done.